homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -207,11 +172,22 @@ def get_router_title(conn: Connection) -> str:
                 user_input=user_input, errors=errors
             )
 
-        title = self.context.get("title_placeholders", {}).get(
-            CONF_NAME
-        ) or await self.hass.async_add_executor_job(get_router_title, conn)
+        info = await self.hass.async_add_executor_job(get_device_info, conn)
         await self.hass.async_add_executor_job(logout)
 
+        if not self.unique_id:
+            if serial_number := info.get("SerialNumber"):
+                await self.async_set_unique_id(serial_number)
+                self._abort_if_unique_id_configured()
+            await self._async_handle_discovery_without_unique_id()
```

Comment:
Should this be in an else ? Because you might have just configured a unique ID.

Reply:
Indeed it should, even though `_async_handle_discovery_without_unique_id` is a no-op if there is a unique id. But will clean up.